### PR TITLE
Fix sparky smartport

### PR DESCRIPTION
**Pull request by ledvinap**
SBUS + SPORT works on Sparky + FrSky X8R (UART2 sbus; UART3 sport, with `set telemetry_inversion = 1`)

### CODE REVIEW

File: src/main/drivers/serial_uart.c [hunk 1]
```diff
@@ -53,17 +53,11 @@ static void usartConfigurePinInversion(uartPort_t *uartPort) {
     uint32_t inversionPins = 0;
 
     // Inversion when using OPTION_BIDIR not supported yet.
```
Comment (thenickdude) [inline src/main/drivers/serial_uart.c]: Probably want to drop the old comment about inversion not being supported!

Comment (ledvinap) [inline src/main/drivers/serial_uart.c]: Yes, I overlooked that. Fixed

**Comment by hydra** [comment]
Ready to merge?

**Comment by ledvinap** [comment]
I'll rewrite the commits if it is ok this way ...

**Comment by hydra** [comment]
Great, I'll merge this shortly.

**Comment by ledvinap** [comment]
Did rewrite, github discarded coments. They are here:
---
@thenickdude 
I don't understand why open-drain is not appropriate. When Tx finishes, the pin is handed back from the UART to GPIO, and the GPIO setting for the pin is PP with this new code. Shouldn't it be Open Drain (i.e. high impedance) so it isn't trying to drive the line while it is reading data from the Rx? Manual says this:

> The TX pin is always released when no data is transmitted. Thus, it acts as a standard I/O in idle or in reception. It means that the I/O must be configured so that TX is configured as alternate function open-drain with an external pull-up.

I don't understand how inversion impacts this port setting. I'm not a GPIO expert so I'd be happy to learn :)

EDIT: Is it because the inversion setting causes the OD to switch to the current sinking state instead of high impedance?

---
ledvinap commented on 447e238 
USART does release pin when not transmitting, so PP works.

Using OD would need pullup to 3.3V, but that will create invalid idle level (idle state is 0V for inverted signal).

In non-inverting case both options will work (I did not test it). Using PP provides better noise immunity, using OD will require pullup, but will handle collisions better.

---
thenickdude commented on 447e238
Got it, thanks!

---
ledvinap commented on 447e238
I may modify code to use OD for non-inverted case ... what is your opinion?

---
thenickdude commented on 447e238 
Yeah I would use it for non-inverted case, simply because the STM32F3 manual recommends it and I figure they know what they're talking about (I don't know anything more than the manual at the moment!).

**Pull request merged**